elf2table/src/lib.rs

```diff
@@ -92,3 +91,187 @@ pub fn parse(elf: &[u8]) -> Result<Option<Table>, anyhow::Error> {
+                                decl_file = Some(idx);
+                            }
+                        }
+
+                        gimli::constants::DW_AT_decl_line => {
+                            if let gimli::AttributeValue::Udata(line) = attr.value() {
+                                decl_line = Some(line);
+                            }
+                        }
+
+                        gimli::constants::DW_AT_location => {
+                            if let gimli::AttributeValue::Exprloc(loc) = attr.value() {
+                                location = Some(loc);
+                            }
+                        }
+
+                        _ => {}
+                    }
+                }
+
+                if name.is_some()
+                    && decl_file.is_some()
+                    && decl_line.is_some()
+                    && location.is_some()
+                {
+                    if let (Some(name_index), Some(file_index), Some(line), Some(loc)) =
+                        (name, decl_file, decl_line, location)
+                    {
+                        let endian_slice = dwarf.string(name_index)?;
+                        let name = core::str::from_utf8(&endian_slice)?;
+
+                        if name == "DEFMT_LOG_STATEMENT" {
+                            let addr = exprloc2address(unit.encoding(), &loc)?;
+                            let file =
+                                PathBuf::from(file_index_to_string(file_index, &unit, &dwarf)?);
+
+                            let loc = Location { file, line };
+
+                            if addr != 0 {
+                                ensure!(
+                                    map.insert(addr, loc).is_none(),
+                                    "BUG in DWARF variable filter: index collision"
+                                );
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    Ok(map)
+}
+
+fn file_index_to_string<R>(
+    index: u64,
+    unit: &gimli::Unit<R>,
+    dwarf: &gimli::Dwarf<R>,
+) -> Result<String, anyhow::Error>
+where
+    R: gimli::read::Reader,
+{
+    ensure!(index != 0, "`FileIndex` was zero");
+
+    let header = if let Some(program) = &unit.line_program {
+        program.header()
+    } else {
+        bail!("no `LineProgram`");
+    };
+
+    let file = if let Some(file) = header.file(index) {
+        file
+    } else {
+        bail!("no `FileEntry` for index {}", index)
+    };
+
+    let mut s = String::new();
+    if let Some(dir) = file.directory(header) {
+        let dir = dwarf.attr_string(unit, dir)?;
+        let dir = dir.to_string_lossy()?;
+
+        if !dir.starts_with('/') {
+            if let Some(ref comp_dir) = unit.comp_dir {
+                s.push_str(&comp_dir.to_string_lossy()?);
+                s.push('/');
+            }
+        }
+        s.push_str(&dir);
+        s.push('/');
```

Comment:
The `/` theoretically would break on Windows when the base path uses `\\?\` UNC paths. Not sure if that's a problem, but might be worth at least a `// FIXME`.